Controller/HeadlessWebsiteController.php

```diff
@@ -75,14 +76,64 @@ private function renderTemplateResponse(
             throw new HttpException(406, sprintf('Page does not exist in "%s" format.', $requestFormat));
         }
 
-        if ($preview) {
-            $parameters['previewParentTemplate'] = $viewTemplate;
-            $parameters['previewContentReplacer'] = Preview::CONTENT_REPLACER;
+        // if partial render only content block else full page
+        if ($partial) {
+            $content = $this->renderBlock(
+                $viewTemplate,
+                'content',
+                $parameters
+            );
+        } elseif ($preview) {
+            $content = $this->renderPreview(
+                $viewTemplate,
+                $parameters
+            );
+        } else {
+            $content = $this->renderView(
+                $viewTemplate,
+                $parameters
+            );
+        }
+
+        return new Response($content);
+    }
+
+    /**
+     * @param mixed[] $parameters
+     */
+    protected function renderBlock(string $template, string $block, array $parameters = []): string
+    {
+        $twig = $this->get('twig');
+        $parameters = $twig->mergeGlobals($parameters);
+
+        $template = $twig->load($template);
 
-            return $this->render('@SuluWebsite/Preview/preview.html.twig', $parameters);
+        $level = ob_get_level();
```

Comment:
this scares me, but i guess we need to support it 😢 where do we use partial rendering?

Reply:
in the preview :)

Reply:
do we really need to use functions like `ob_get_level` for this? 😕